### PR TITLE
add docs for intentOptions

### PR DESCRIPTION
**Pull request by rajdeepv**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/en/writing-running-appium/android/android-launch-intent-options.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+## How To Specify App Launch Intent Options
+
+Most of the times Appium users do not need to worry about how app is launched as Appium takes care of this internally.
+However, sometimes a user need the flexibility to control it. Eg. when a user need to pass some extra
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/android/android-launch-intent-options.md]: needs

Comment (mykola-mokhnach) [reply docs/en/writing-running-appium/android/android-launch-intent-options.md]: E.g.

Comment (mykola-mokhnach) [reply docs/en/writing-running-appium/android/android-launch-intent-options.md]: to the app

**Comment by KazuCocoa** [comment]
Could you add this md file in toc.js like https://github.com/appium/appium/blob/master/docs/toc.js#L252 to add this in appium.io?

File: docs/en/writing-running-appium/android/android-launch-intent-options.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+## How To Specify App Launch Intent Options
+
+Most of the times Appium users do not need to worry about how app is launched as Appium takes care of this internally.
+However, sometimes a user need the flexibility to control it. Eg. when a user need to pass some extra 
+arguments to app during startup which can be used to modify app's behaviour
+
+###How does the appium start App under test**
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/android/android-launch-intent-options.md]: How does Appium start the app...

File: docs/en/writing-running-appium/android/android-launch-intent-options.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+**Appium-UiAutomator2-Driver:** The Appium-UiAutomator2-Driver starts the app under test by issuing an intent using adb command ([read more about intents](https://developer.android.com/reference/android/content/Intent.html)). This adb command
+is invoked by javascript driver. Appium users can pass extra arguments to this adb command using capabilitie `intentAction` ,`intentCategory`,`intentFlags` and `optionalIntentArguments`
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/android/android-launch-intent-options.md]: capabilities

File: docs/en/writing-running-appium/android/android-launch-intent-options.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+**Appium-Espresso-Driver:**
+The Appium-Espresso-Driver starts the the app by issuing the intent using the kotlin server. For starting app a default intent is used.
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/android/android-launch-intent-options.md]: using the server written in Kotlin

Comment (mykola-mokhnach) [reply docs/en/writing-running-appium/android/android-launch-intent-options.md]: An intent with default options is used to start the app.

File: docs/en/writing-running-appium/android/android-launch-intent-options.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+A user can override this default intent using `intentOptions` capability,
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/android/android-launch-intent-options.md]: ...these default intent options using...

File: docs/en/writing-running-appium/android/android-launch-intent-options.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+###How does the appium start App under test**
+
+**Appium-UiAutomator2-Driver:** The Appium-UiAutomator2-Driver starts the app under test by issuing an intent using adb command ([read more about intents](https://developer.android.com/reference/android/content/Intent.html)). This adb command
+is invoked by javascript driver. Appium users can pass extra arguments to this adb command using capabilitie `intentAction` ,`intentCategory`,`intentFlags` and `optionalIntentArguments`
+
+**Appium-Espresso-Driver:**
+The Appium-Espresso-Driver starts the the app by issuing the intent using the kotlin server. For starting app a default intent is used.
+A user can override this default intent using `intentOptions` capability, 
+
+eg in ruby, you can specify `intentOptions` capability as below:
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/android/android-launch-intent-options.md]: E.g.

File: docs/en/writing-running-appium/android/android-launch-intent-options.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+````
+intentOptions => {
+    'action' => 'android.intent.action.MAIN',
+    'categories' => 'android.intent.category.LAUNCHER',
+    'component' => 'com.my.app/.MyLauncherActivity',
+    'flags' => 'FLAG_ACTIVITY_NEW_TASK',
+    'es' => [' key', ' Value ']
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/android/android-launch-intent-options.md]: Spaces in quotes are redundant

File: docs/en/writing-running-appium/android/android-launch-intent-options.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+**Appium-UiAutomator2-Driver:** The Appium-UiAutomator2-Driver starts the app under test by issuing an intent using adb command ([read more about intents](https://developer.android.com/reference/android/content/Intent.html)). This adb command
+is invoked by javascript driver. Appium users can pass extra arguments to this adb command using capabilitie `intentAction` ,`intentCategory`,`intentFlags` and `optionalIntentArguments`
+
+**Appium-Espresso-Driver:**
+The Appium-Espresso-Driver starts the the app by issuing the intent using the kotlin server. For starting app a default intent is used.
+A user can override this default intent using `intentOptions` capability, 
+
+eg in ruby, you can specify `intentOptions` capability as below:
+
+````
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/android/android-launch-intent-options.md]: three backticks would be enough

File: docs/en/writing-running-appium/android/android-launch-intent-options.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    'action' => 'android.intent.action.MAIN',
+    'categories' => 'android.intent.category.LAUNCHER',
+    'component' => 'com.my.app/.MyLauncherActivity',
+    'flags' => 'FLAG_ACTIVITY_NEW_TASK',
+    'es' => [' key', ' Value ']
+}
+                                    
+````
+
+If you are using `intentOptions`, it is advisable to provide `action`, `categories`, `component` and `flags`. Value of `flags` must include `FLAG_ACTIVITY_NEW_TASK` if a new activity from outside of current context is being started
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/android/android-launch-intent-options.md]: I think it makes sense to provide full description of all available intent options here

**Pull request closed without merging**